spec/mem/pointer.md

```diff
@@ -43,32 +43,32 @@ impl<Provenance> Pointer<Provenance> {
 We sometimes need information what it is that a pointer points to, this is captured in a "pointer type".
 
 ```rust
-/// A "layout" describes what we know about data behind a pointer.
-pub struct Layout {
+/// Describes what we know about data behind a pointer.
+/// (This also means we have a finite representation even when the Rust type is recursive.)
```

Comment:
This comment should go into `types.md`, not here.

Reply:
My suggestion would be to add it to the `Ptr(PtrType)` variant:
```rust
/// `Ptr` represents all pointer types: references, raw pointers, boxes, and function pointers.
/// A pointer type does *not* need the full pointee type, since (de)serializing a pointer does not
/// require knowledge about the pointee. We only track basic pointee information like size and
/// alignment that is required to check reference validity. This also means types have a finite
/// representation even when the Rust type is recursive.
Ptr(PtrType),
```

Reply:
Thanks :)